casper/src/main/resources/PoS.rhox

```diff
@@ -0,0 +1,164 @@
+// Rholang macro parameters:
+// minimumBond - the minimum bond allowed by the PoS
+// maximumBond - the maximum bond allowed by PoS
+
+/*
+The table below describes the required computations and their dependencies
+
+No. | Dependency | Computation method | Result
+----+------------+--------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------
+1.  |            | given              | sk = d70491c21a5ba44ffb60a97d9bfbf3f3839b997bde107a3a8ab6872594282740
+2.  |            | given              | timestamp = 1553862086627
+3.  |            | lastNonce          | nonce = 9223372036854775807
+4.  | 1,         | ed25519            | pk = cc87bf7747a8c176714b417ca14a63897b07572876c5e38a7896b6007738ef81
+5.  | 4, 2,      | genIds             | uname = Unforgeable(0xf974f799067b34be1acb7b565e0b160edb23df3e3ad58f200af183ae48b458cd)
+6.  | 3, 5,      | registry           | value = (9223372036854775807, bundle+ {   Unforgeable(0xf974f799067b34be1acb7b565e0b160edb23df3e3ad58f200af183ae48b458cd) })
+7.  | 6,         | protobuf           | toSign = 2a3eaa013b0a0d2a0b10feffffffffffffffff010a2a5a280a243a220a20f974f799067b34be1acb7b565e0b160edb23df3e3ad58f200af183ae48b458cd1001
+8.  | 7, 1,      | ed25519            | sig = 7ff160accda7f76f14f2ae132aaae140ac22b2671e561a58a566bff1a01e49cf372201a5df678aba5ff09171b6af966906deffbf8933efeeddb69a4c204b0503
+9.  | 4,         | registry           | uri = rho:id:cnec3pa8prp4out3yc8facon6grm3xbsotpd4ckjfx8ghuw77xadzt
+----+------------+--------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------
+*/
+
+new PoS,
+    registryLookup(`rho:registry:lookup`),
+    revAddressOps(`rho:rev:address`),
+    revVaultCh,
+    getUser,
+    getCurrentUserAddress,
+    getCurrentUserVault,
+    getMVar,
+    runMVar,
+    deposit in {
+  registryLookup!(`rho:id:1o93uitkrjfubh43jt19owanuezhntag5wh74c6ur5feuotpi73q8z`, *revVaultCh) |
+
+  for(@(_, RevVault) <- revVaultCh) {
+    new posRevAddressCh in {
+
+      revAddressOps!("fromPublicKey", "cc87bf7747a8c176714b417ca14a63897b07572876c5e38a7896b6007738ef81".hexToBytes(), *posRevAddressCh) |
+
+      for(@posRevAddress <- posRevAddressCh) {
+        new bondsCh in {
+          bondsCh!({}) |
+          getCurrentUserVault!(Nil) |
```

Comment:
We should not use Nil as a return channel - it's going to linger there. Also, what's the purpose of this call?

Reply:
If it is creation of the PoS vault, let's just do `findOrCreate!(revAddress)` directly without resolving the address yet again

Reply:
are you saying that this call is not really needed and the vault will be created when it's used the first time?

Reply:
I'm saying let's use the interpolated posVaultRevAddress to spawn the vault, instead of inferring the revAddress from the pubkey of the deployer of this system contract.

No big deal though, I think - it should not be a public-key-bound vault anyway.

Here's a ticket for having that API https://rchain.atlassian.net/browse/RCHAIN-3427

And here's one for using that API https://rchain.atlassian.net/browse/RCHAIN-3428

If you could pull in some of those tickets and fix a couple of them in a single PR we'd be able to avoid some churn :)

Reply:
let's do it simple, like it is implemented now and we'll figure out the final solution as we go. Those two tickets are not defined yet and there is no need to postpone the merge of this PR